index-scheduler/src/lib.rs

```diff
@@ -606,6 +676,25 @@ impl IndexScheduler {
 
         Ok(processed_tasks)
     }
+
+    pub(crate) fn delete_persisted_task_data(&self, task: &Task) -> Result<()> {
+        match &task.kind {
+            KindWithContent::DocumentImport { content_file, .. } => {
+                self.delete_update_file(*content_file)
+            }
+            KindWithContent::DocumentDeletion { .. }
+            | KindWithContent::DocumentClear { .. }
+            | KindWithContent::Settings { .. }
+            | KindWithContent::IndexDeletion { .. }
+            | KindWithContent::IndexCreation { .. }
+            | KindWithContent::IndexUpdate { .. }
+            | KindWithContent::IndexSwap { .. }
+            | KindWithContent::TaskCancelation { .. }
+            | KindWithContent::TaskDeletion { .. }
+            | KindWithContent::DumpExport { .. }
+            | KindWithContent::Snapshot => Ok(()),
+        }
+    }
```

Comment:
```suggestion
    if let Some(content_file) = task.content_uuid() {
        self.delete_update_file(*content_file)?;
    }
    Ok(())
```

We have a function for that